### PR TITLE
Feature/optimize chatroom ui

### PR DESCRIPTION
**Pull request by Jerome1210**

### CODE REVIEW

File: Examples/VideoChat/ChatRoom/ChatViewController.m [hunk 1]
```diff
@@ -36,6 +36,13 @@ @interface ChatViewController ()
 @property (nonatomic, getter=hasUpdatedNickname) BOOL updatedNickname;
 @property (nonatomic) NSString * fakeName;
 
+//Custom view
+@property (nonatomic) UIActivityIndicatorView * indicator;
+@property (nonatomic) NSMutableArray * allLayoutConstraints;
+
+@property (nonatomic) NSMutableArray * cachedAddMessages;
+@property (nonatomic) NSMutableArray * cachedRemoveMessageIds;
```
Comment (lollyrain) [inline Examples/VideoChat/ChatRoom/ChatViewController.m]: How about `cachedAddedMessages` ＆ `cachedRemovedMessageIds`?

File: Examples/VideoChat/ChatRoom/ChatViewController.m [hunk 2]
```diff
@@ -618,6 +630,48 @@ - (void)addFakeMessage:(NSString *)fakeMessage type:(STSChatMesssageType)type im
...
 };
 
+#pragma mark - update tableView timer
+
+- (void)startUpdateTableViewTimer {
+    self.updateTableViewTimer = [NSTimer scheduledTimerWithTimeInterval:self.refreshTableViewTimeInteval
+                                                                 target:self
+                                                               selector:@selector(updateTabelViewTimerFired)
+                                                               userInfo:nil
+                                                                repeats:YES];
```
Comment (Jerome1210) [inline Examples/VideoChat/ChatRoom/ChatViewController.m]: thanks to point out this, fix 991498d

File: Examples/VideoChat/ChatRoom/ChatViewController.h [hunk 3]
```diff
@@ -60,6 +60,12 @@
 @property (nonatomic) BOOL autoConnect;
 
 /**
+ A boolean value indicates whether to add default activityIndicatorView at chatViewController's view.
+ Default is YES, if you want to add your custom loading view set it false.
+ */
+@property (nonatomic) BOOL shouldAddIndicatorView;
```
Comment (Jerome1210) [inline Examples/VideoChat/ChatRoom/ChatViewController.h]: change e65f233

File: Examples/VideoChat/ChatRoom/ChatViewController.m (hunk 1 above)
Comment (Jerome1210) [inline Examples/VideoChat/ChatRoom/ChatViewController.m]: change fa2ce37

**Pull request merged**